src/cockpit/389-console/src/lib/ldap_editor/wizards/operations/addCosDefinition.jsx

```diff
@@ -0,0 +1,1053 @@
+                            <Alert
+                                variant={namingAttr === '' || namingVal === ''
+                                    ? 'warning'
+                                    : 'success'}
+                                isInline
+                                title={myTitle}
+                            >
+                                <b>Entry DN:&nbsp;&nbsp;&nbsp;</b>{(namingAttr ? namingAttr : "??????")}={namingVal ? namingVal : "??????"},{this.props.wizardEntryDn}
+                            </Alert>
+                        </GridItem>
+                        <GridItem span={12} className="ds-margin-top-xlg">
+                            <TextContent>
+                                <Text component={TextVariants.h3}>
+                                    Set Attribute Values
+                                </Text>
+                            </TextContent>
+                        </GridItem>
+                    </Grid>
+                </Form>
+                <GridItem className="ds-left-margin" span={11}>
+                    <EditableTable
+                        editableTableData={editableTableData}
+                        isAttributeSingleValued={this.isAttributeSingleValued}
+                        isAttributeRequired={this.isAttributeRequired}
+                        enableNextStep={this.enableNextStep}
+                        setNamingRowID={this.setNamingRowID}
+                        saveCurrentRows={this.saveCurrentRows}
+                        namingRowID={this.state.namingRowID}
+                        namingAttr={this.state.namingAttr}
+                    />
+                </GridItem>
+            </React.Fragment>
+        );
+
+        const ldifListItems = ldifArray.map((line, index) =>
+            <SimpleListItem key={index} isCurrent={line.startsWith('dn: ')}>
+                {line}
+            </SimpleListItem>
+        );
+
+        const cosCreationStep = (
+            <div>
+                <Alert
+                    variant="info"
+                    isInline
+                    title="LDIF Content for CoS Creation"
+                />
+                <Card isHoverable>
+                    <CardBody>
+                        {(ldifListItems.length > 0) &&
+                            <SimpleList aria-label="LDIF data User">
+                                {ldifListItems}
+                            </SimpleList>
+                        }
+                    </CardBody>
+                </Card>
+            </div>
+        );
+
+        let nb = -1;
+        const ldifLines = ldifArray.map(line => {
+            nb++;
+            return { data: line, id: nb };
+        })
+        const cosReviewStep = (
+            <div>
+                <Alert
+                    variant={resultVariant}
+                    isInline
+                    title="Result for CoS Creation"
+                >
+                    {commandOutput}
```

Comment:
I started adding a spinner to this alert, because when you have a large db the rendering of the update and rebuilding of the suffix tree can take a long time (20+ seconds>, and it looks like the UI is hung.  Look at addGroup.jsx line 576:

```
                    {this.state.adding &&
                        <div>
                            <Spinner className="ds-left-margin" size="md" />
                            &nbsp;&nbsp;Adding group ...
                        </div>
                    } 
```

You'll need to add other code to handle "this.state.adding", but its pretty straight forward in addGroup.jsx